src/wallet/mod.rs

```diff
@@ -5414,4 +5417,40 @@ pub(crate) mod test {
         // ...and checking that everything is fine
         assert_fee_rate!(psbt, details.fee.unwrap_or(0), fee_rate);
     }
+
+    #[cfg(feature = "test-hardware-signer")]
+    #[test]
+    fn test_create_signer() {
+        use hwi::{types, HWIClient};
+
+        use crate::wallet::hardwaresigner::HWISigner;
+
+        let devices = HWIClient::enumerate().unwrap();
+        let client = HWIClient::get_client(
+            devices.first().expect(
+                "No devices found. Either plug in a hardware wallet, or start a simulator.",
+            ),
+            true,
+            types::HWIChain::Regtest,
+        )
+        .unwrap();
+        let descriptors = client.get_descriptors(None).unwrap();
+        let custom_signer =
+            HWISigner::from_device(devices.first().unwrap(), types::HWIChain::Regtest).unwrap();
```

Comment:
nit: you could save `devices.first().expect("...")` (above) in a variable, and use it here as well